service/src/main/java/bio/terra/cbas/controllers/ControllerUtils.java

```diff
@@ -0,0 +1,31 @@
+package bio.terra.cbas.controllers;
+
+import bio.terra.cbas.dependencies.sam.SamService;
+import bio.terra.cbas.util.BearerTokenFilter;
+import java.util.Optional;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+
+@Component
+public class ControllerUtils {
+  private final SamService samService;
+
+  @Autowired
+  ControllerUtils(SamService samService) {
+    this.samService = samService;
+  }
+
+  Optional<String> getUserToken() {
+    Object token =
+        RequestContextHolder.currentRequestAttributes()
```

Comment:
Since I had to go look this up, I would maybe add a comment here along the lines of (stolen from the javadoc):
```
Exposes the web request in the form of a thread-bound [RequestAttributes](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/context/request/RequestAttributes.html) object.
```

So as long as we call this function from the _thread_ that's servicing a request, we should be certain to get the right request's headers. Which is **_great_**, that was the main thing I was afraid of.